pkg/azureutils/logger.go

```diff
@@ -0,0 +1,189 @@
+package azureutils
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2/klogr"
+	diskv1beta1 "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1beta1"
+)
+
+type contextKey int
+
+const workflowKey contextKey = 1
```

Comment:
I believe you could use an empty struct of unique type for the key. This is what [`logr.NewContext()`](https://github.com/go-logr/logr/blob/47e013cee9b1f91c987cc70a218639655431b607/logr.go#L395) does.